### PR TITLE
Book crud functions

### PR DESCRIPTION
**Pull request by cwahl02**
This pull request introduces robust support for Data Transfer Objects (DTOs) for both `Account` and `Book` entities, refactors repository update methods to use DTOs with validation, and adds comprehensive unit tests for the new repository logic. These changes improve code maintainability, enable partial updates, and ensure data integrity through validation.

### DTO and Entity Model Enhancements

* Added `AccountUpdateDTO` and `BookUpdateDTO` classes to support partial and validated updates for `Account` and `Book` entities in `BookClub/Models/Account.cs` and `BookClub/Models/Book.cs`. [[1]](diffhunk://#diff-3d18d3d5aef98d7768929a9fe94af7bae605de4de0d1f7fd31f411ae719aa410R38-R60) [[2]](diffhunk://#diff-9a46f222dadab1cf5932b0ad551c94f04b663810762e5c982da421c0e7a0dd1dR39-R53)
* Updated entity model documentation with XML comments for clarity in `BookClub/Models/Book.cs`.

### Repository Refactoring and Validation

* Refactored `AccountsRepository.Update` to accept an account ID and `AccountUpdateDTO`, apply validation, and support partial updates.
* Implemented `BookRepository` with CRUD operations, including update and delete methods that use `BookUpdateDTO` and validation.
* Added null checks and validation to repository methods to prevent invalid operations. [[1]](diffhunk://#diff-a90539e7a8111dea1a16edfddf60ba6bc71844bbc320e5093324c18b953a6a1cR40-R42) [[2]](diffhunk://#diff-210c417cf923619944a826b7d51ca1ab44d4965f2480e3c599efa24fea304a8eR1-R76)

### UI and Integration Updates

* Updated password reset logic in `ResetPassword.cs` to use the new DTO-based update method for accounts.

### Testing Improvements

* Refactored and expanded unit tests for `AccountsRepository` to cover new DTO-based update logic and edge cases, improving test reliability and maintainability. [[1]](diffhunk://#diff-e9ac916906ad434f345387b3e98a2eb87a3866f85097f3e3df8b0fc5f9fe1f1aR19-R21) [[2]](diffhunk://#diff-e9ac916906ad434f345387b3e98a2eb87a3866f85097f3e3df8b0fc5f9fe1f1aL27-R81) [[3]](diffhunk://#diff-e9ac916906ad434f345387b3e98a2eb87a3866f85097f3e3df8b0fc5f9fe1f1aR93) [[4]](diffhunk://#diff-e9ac916906ad434f345387b3e98a2eb87a3866f85097f3e3df8b0fc5f9fe1f1aL70-R190)
* Added comprehensive unit tests for `BookRepository`, covering all CRUD operations and validation scenarios.

### CODE REVIEW

File: BookClub/Repositories/BookRepository.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        var validationContext = new ValidationContext(dto);
+        var validationResults = new List<ValidationResult>();
+
+        bool isValid = Validator.TryValidateObject(dto, validationContext, validationResults, true);
+
+        if (!isValid)
+        {
+            var errors = string.Join("; ", validationResults.Select(r => r.ErrorMessage));
+            MessageBox.Show($"BookUpdateDTO validation failed:\n{errors}");
```
[the remark this answers is not in the document]
Comment (CW-ISM) [reply BookClub/Repositories/BookRepository.cs:52]: This is true, should return string or use `out` an out variable. A future concern though is if we make CRUD functions async then we can't use an `out` variable
